bears/general/SpaceConsistencyBear.py

```diff
@@ -38,7 +41,36 @@ def run(self,
         result_texts = []
         additional_info_texts = []
 
-        for line_number, line in enumerate(file, start=1):
+        line_nr_start = next((i for i, line in reversed(list(
+            enumerate(file, start=1))) if line.strip() != ''), 0)
+
+        if len(file) == 0:
+            blanklines_start_line = False
+
+        else:
+            blanklines_start_line = line_nr_start + 1 \
+                    if line_nr_start < len(file) else False
```

Comment:
I have the feeling you are unnecessary trying to use special values (--> `False`). 

You can easily check if you have to yield a result if `line_nr_start < len(file)`.
The whole stuff can be refactored to roughly this:
```python
if not allow_trailing_blanklines:
    line_nr_start = next((i for i, line in reversed(list(
        enumerate(file, start=1))) if line.strip() != ''), 0)
    # TODO: Maybe more performance: Don't increment from start=1, but start=0 and
    #       the next-default value = -1, and increment in the end: next(..., -1) + 1

    if line_nr_start < len(file):
        # Note: You have to provide line_nr_start + 1 to refer to the right lines.
        # line_nr_start is right now designed to give the starting line referring
        # to zero-based indices.
        yield Result(...)
else:
    line_nr_start = len(file)

for line_number, line in enumerate(file[:line_nr_start])
```

Reply:
Note also how I've put the relevant analysis inside the if, to save performance in case someone wants not to take care of blanklines

Reply:
yeah, I wanted to do the same.But I was finding it difficult to cover the case when file is empty.That's why I used that (--> False), but now it will work as we are using `start=0` and next default value `-1`.